exporter/kafkaexporter/exporter.go

```diff
@@ -0,0 +1,102 @@
+package kafkaexporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"sync"
+	"text/template"
+
+	"github.com/IBM/sarama"
+	"github.com/thomaspoignant/go-feature-flag/exporter"
+	"github.com/thomaspoignant/go-feature-flag/utils/fflog"
+)
+
+type Exporter struct {
+	// Format is the output format you want in your exported file.
+	// Available format are JSON, CSV and Parquet.
+	// Default: JSON
+	Format string
+
+	// CsvTemplate is used if your output format is CSV.
+	// This field will be ignored if you are using another format than CSV.
+	// You can decide which fields you want in your CSV line with a go-template syntax,
+	// please check internal/exporter/feature_event.go to see what are the fields available.
+	// Default:
+	// {{ .Kind}};{{ .ContextKind}};{{ .UserKey}};{{ .CreationDate}};{{ .Key}};{{ .Variation}};{{ .Value}};{{ .Default}}\n
+	CSVTemplateString string
+
+	Settings Settings
+
+	sender      MessageSender
+	init        sync.Once
+	csvTemplate *template.Template
+}
+
+// Export is saving a collection of events in a file.
+func (f *Exporter) Export(_ context.Context, logger *log.Logger, featureEvents []exporter.FeatureEvent) error {
+	if f.sender == nil {
+		err := f.initializeWriter()
+		if err != nil {
+			return fmt.Errorf("writer: %w", err)
+		}
+	}
+
+	var messages []*sarama.ProducerMessage
+	for _, event := range featureEvents {
+		data, err := f.formatMessage(event)
+		if err != nil {
+			return fmt.Errorf("format: %w", err)
+		}
+
+		messages = append(messages, &sarama.ProducerMessage{
+			Topic: f.Settings.Topic,
+			Key:   sarama.StringEncoder(event.UserKey),
+			Value: sarama.ByteEncoder(data),
+		})
+	}
+
+	err := f.sender.SendMessages(messages)
+	if err != nil {
+		return fmt.Errorf("send: %w", err)
+	}
+
+	fflog.Printf(logger, "info: [KafkaExporter] sent %d messages", len(messages))
+	return nil
+}
+
+func (f *Exporter) IsBulk() bool {
+	return true
```

Comment:
To answer your question 
>Async vs Sync: Kafka is designed to work in either synchronous or asynchronous mode, and the Kafka library I've chosen supports both. My understanding is that by design the exporters will always be called asynchronously, and thus the message publishing should be synchronous as doing it asynchronously would be redundant. Is this assumption correct?

If you do this change

```suggestion
	return false
```

The exporter will not be async anymore and will send every event 1 by 1 to the exporter.
This is what we are doing also in the SQS exporter _([see here](https://github.com/thomaspoignant/go-feature-flag/blob/fd90988686cccb85f23cdedc39f7e2bc65c47380/exporter/sqsexporter/exporter.go#L74))_.

Since Kafka is designed to receive the events 1 by 1, I think we need to set this to false.